astropy/modeling/core.py

```diff
@@ -1063,11 +1065,16 @@ def _validate_input_shapes(self, inputs, argnames, model_set_axis):
             )
 
         try:
-            input_shape = check_broadcast(*all_shapes)
-        except IncompatibleShapeError as e:
-            raise ValueError(
-                "All inputs must have identical shapes or must be scalars."
-            ) from e
+            input_shape = np.broadcast_shapes(*all_shapes)
+        except ValueError as exc:
+            note = "All inputs must have identical shapes or must be scalars."
+            if sys.version_info >= (3, 11):
+                exc.add_note(note)
+            else:
+                # mimic Python 3.11 behavior:
+                # preserve error message and traceback
+                exc = type(exc)(f"{exc!s}\n{note}").with_traceback(exc.__traceback__)
+            raise exc
```

Comment:
This is a side improvement but isn't strictly needed here, I just applied an existing snippet of mine here. Let me know if I should revert it or move it to a separate PR.